storagetest/storagetest.go

```diff
@@ -2,69 +2,232 @@ package storagetest
 
 import (
 	"context"
+	"fmt"
+	"path"
+	"sort"
+	"strings"
+	"sync/atomic"
 	"testing"
 
 	"github.com/giantswarm/microstorage"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 // Test is Storage conformance test.
 func Test(t *testing.T, storage microstorage.Storage) {
```

Comment:
I apparently miss to understand how this test works. Where is `Test` called? I think in general the style of testing is too magic to me personally. Somebody once said "Be clear, not clever". I think when you look at a test you should be able to understand it by reading it from top to bottom. This almost never be true but my problem here is that I have to check different files and lookup a lot of different methods. It is also that all tests are executed in the same golang test. This does also not feel right to me. There should be one test function testing one thing, expect of table driven tests where you test one thing using different input variations.